File: src/系统/09．表现系统/02．对话框系统入口/03．配置查询.lua
local ____lualib = require("lualib_bundle")
local __TS__ArrayFind = ____lualib.__TS__ArrayFind
local ____exports = {}
local ____01_FF0E_5BF9_8BDD_914D_7F6E_8868 = require("系统.08．任务系统.00．配置表.01．对话配置表")
local DIALOG_NPC_CONFIGS = ____01_FF0E_5BF9_8BDD_914D_7F6E_8868.DIALOG_NPC_CONFIGS
local ____03_FF0ENPC_914D_7F6E_8868 = require("系统.08．任务系统.00．配置表.03．NPC配置表")
local NPC_CONFIGS = ____03_FF0ENPC_914D_7F6E_8868.NPC_CONFIGS
local ____02_FF0E_4EFB_52A1_914D_7F6E_8868 = require("系统.08．任务系统.00．配置表.02．任务配置表")
local QUEST_CONFIGS = ____02_FF0E_4EFB_52A1_914D_7F6E_8868.QUEST_CONFIGS
function ____exports.findQuestByNpc(self, npcName)
    return __TS__ArrayFind(
        QUEST_CONFIGS,
        function(____, quest) return quest.startNpc == npcName and quest.requireID end
    )
end
function ____exports.findDialogConfig(self, npcName)
    return __TS__ArrayFind(
        DIALOG_NPC_CONFIGS,
        function(____, config) return config.NPC == npcName end
    )
end
function ____exports.findNpcConfigByUnitName(self, unitName)
    for ____, npc in ipairs(NPC_CONFIGS) do
        if npc.NPCrequireName == unitName or npc.NpcNameID == unitName then
            return npc
        end
    end
    return nil
end
return ____exports

File: src/系统/09．表现系统/02．对话框系统入口/03．配置查询.js
import { DIALOG_NPC_CONFIGS } from "../../08．任务系统/00．配置表/01．对话配置表";
import { NPC_CONFIGS } from "../../08．任务系统/00．配置表/03．NPC配置表";
import { QUEST_CONFIGS } from "../../08．任务系统/00．配置表/02．任务配置表";
import { hasPlayerAcceptedQuest } from "./02．任务状态";
// ========== 虚拟分区：配置查询 ==========
export function findQuestByNpc(npcName) {
    return QUEST_CONFIGS.find(quest => quest.enabled === true && quest.startNpc === npcName && quest.requireID);
}
export function resolveQuestEndNpc(quest) {
    const endNpc = quest.endNpc;
    if (!endNpc || endNpc === "没有")
        return quest.startNpc || "";
    return endNpc;
}
export function findAcceptedQuestBySubmitNpc(npcName, playerId) {
    return QUEST_CONFIGS.find(quest => {
        if (quest.enabled !== true)
            return false;
        if (!quest.requireID)
            return false;
        const questId = quest.requireID.toString();
        if (!hasPlayerAcceptedQuest(playerId, questId))
            return false;
        return resolveQuestEndNpc(quest) === npcName;
    });
}
export function findDialogConfig(npcName) {
    return DIALOG_NPC_CONFIGS.find(config => config.NPC === npcName);
}
export function findNpcConfigByUnitName(unitName) {
    for (const npc of NPC_CONFIGS) {
        if (npc.NPCrequireName === unitName || npc.NpcNameID === unitName)
            return npc;
    }
    return null;
}
